### PR TITLE
Individual listing UI

### PR DESCRIPTION
**Pull request by Pochetes**
## Highlights 🥇 (End of Milestone 1)
- the user can now click on any listing and see more details about it as well as a reservations card (still v1)
- made custom hook for `ListingDetail`, which holds logic for fetching a single listing by id and reservation submission
- added api model functions from `reservations` endpoint in `apiClient.js` which allows user to 
    - see reservations already made for that listing
    - create reservation
    - see the reservations they have made as a user
    - the reservations others have made to their created listings.

<a href="https://www.loom.com/share/4af18b3d0c404e9db2a21f7cb100565e">
    
  </a>

---
`NOTE` In the listing UI, I have included hardcoded information about the dimensions of the pool like the length width (inches) and depth (feet). Looking to implement that in milestone 2 😄 

For next time, would you prefer I leave out the future implementations that are half-done? and only include them in the PR when they're fully implemented (e.g. error, isProcessing, depth, width, length, multi images)

### CODE REVIEW

File: swimster-ui/src/hooks/useListingDetail.js [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+    // handler for making a reservation for the current listing
+    const onSubmit = async formData => {
+        setIsProcessing(true)
+
+        const formattedFormData = {
+            reservationDate: formData.reservationDate,
+            startTime: formData.reservationStartTime,
+            endTime: formData.reservationEndTime,
+            guests: 4, // value to be changed
```
Comment (roeshanigos) [inline swimster-ui/src/hooks/useListingDetail.js:41]: for hygiene it's nice to prefix to do items with

`TODO: value to be changed`


so you dont forget about it (helpful when you start to have many files you can just Biggrep!)

File: swimster-ui/src/components/Navbar/Navbar.jsx [hunk 2]
```diff
@@ -2,15 +2,18 @@ import * as React from 'react'
...
+import { useNavigate } from 'react-router-dom'
 import { style } from './style'
 
 const Navbar = () => {
     const { handlers: authHandlers, user } = useAuthContext()
     const { handlers: listingsHandlers } = useListingsContext()
+    const navigate = useNavigate()
 
     const handleFullLogout = () => {
         // listingsHandlers.clearListings()
```
Comment (roeshanigos) [inline swimster-ui/src/components/Navbar/Navbar.jsx:14]: nit: stray comment? or for next pr?

File: swimster-ui/src/hooks/useListingDetail.js [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+import { useAuthContext, isUserAuthenticated } from "../contexts/auth"
+import { useNavigate } from 'react-router-dom'
+import { useForm } from 'react-hook-form'
+import apiClient from "../services/apiClient"
+
+export const useListingDetail = listingId => {
+    const { user, initialized } = useAuthContext()
+    const { register, handleSubmit } = useForm()
+    const [isFetching, setIsFetching] = useState(false)
+    const [isProcessing, setIsProcessing] = useState(false)
```
Comment (roeshanigos) [inline swimster-ui/src/hooks/useListingDetail.js:11]: might be good to be more specific

`isSubmitProcessing`

File: swimster-ui/src/components/App/App.jsx [hunk 4]
```diff
@@ -32,17 +33,14 @@ const App = () => {
...
   return (
     <BrowserRouter>
       <main className={style.app}>
         <Navbar />
         <Routes>
-          <Route path='/' element={
-            <ProtectedRoute element={<Home />} />
-          }
-          />
+          <Route path='/' element={<Home />} />
```
Comment (roeshanigos) [inline swimster-ui/src/components/App/App.jsx:43]: why did you take the protected route out?

**Review by roeshanigos — APPROVED** [review]
As for your last question, there is no preference as long as you mark it as being implemented soon in a comment with a `TODO:` prefix!

File: swimster-ui/src/hooks/useListingDetail.js (hunk 1 above)
Comment (Pochetes) [reply swimster-ui/src/hooks/useListingDetail.js:41]: that's a good idea, will do!

File: swimster-ui/src/components/Navbar/Navbar.jsx (hunk 2 above)
Comment (Pochetes) [reply swimster-ui/src/components/Navbar/Navbar.jsx:14]: yeah next PR, will modify it to `TODO`

File: swimster-ui/src/hooks/useListingDetail.js (hunk 3 above)
Comment (Pochetes) [reply swimster-ui/src/hooks/useListingDetail.js:11]: done!

File: swimster-ui/src/components/App/App.jsx (hunk 4 above)
Comment (Pochetes) [reply swimster-ui/src/components/App/App.jsx:43]: It was giving me a bug when rendering listings on first load, referenced in this [commit](https://github.com/Facebook-University-Projects/Swimster/pull/9/commits/e7c680f9ac69603a941fa5158dc58a8a20c80439)

Comment (Pochetes) [reply swimster-ui/src/components/App/App.jsx:43]: I was able to solve the issue by redirecting non-logged in users to `/login`, like so
https://github.com/Facebook-University-Projects/Swimster/blob/e67c4a57c5f1de64ad7981c04cf9021410b2cc77/swimster-ui/src/components/Navbar/Navbar.jsx#L13-L17

**Pull request merged**